code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -649,6 +649,16 @@ func (pr *PriceRange) isMatch(price currency.Coin) bool {
 	return pr.Min <= price && price <= pr.Max
 }
 
+// File access control options for StorageAllocation
+type AllocationFileOptions struct {
+	Upload bool `json:"upload"`
+	Delete bool `json:"delete"`
+	Update bool `json:"update"`
+	Move   bool `json:"move"`
+	Copy   bool `json:"copy"`
+	Rename bool `json:"rename"`
```

Comment:
We can use a `int8` to represent the file options in binary format to save spaces. So for example,

```go
0000,0001 - upload 
0000,0010 - delete 
0000,0100 - update
0000, 1000 - move
0001,0000 - copy
0010,0000 - rename 
```

and we can use `&` to check if specific position is set, such as to check if the `update` is set, we can
```go
var allocFileOption int8
if allocFileOption&0000100 > 0 {
   
}

```

We can add the value up for combining different options, so if we want set both`upload` and `delete` to true, we can simply add them:

```go
op := 000001+000010
```

Reply:
Well yeah, much better and optimal. Can try to implement :)

Reply:
I'll check shortly